File: Backend/src/main/java/Gruppe/C/Backend/ChessBot/ChessBot.java
package Gruppe.C.Backend.ChessBot;

import Gruppe.C.Backend.User.User;
import com.github.bhlangonijr.chesslib.move.Move;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.LinkedList;
import java.util.List;

@Data
@Entity
public class ChessBot {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long chessBotId;

    @JoinColumn(name = "userid", referencedColumnName = "id")
    @ManyToOne
    private User player;

    private String chessGameName;

    private Integer timerUser;

    private boolean started;

    private boolean finished;

    private boolean playerInGame;

    private String boardState;

    private byte level; // 1, 2 oder 3

    private boolean turn;  // true -> User, false -> Bot

    private String winner; // Soll nur "Bot", "User" oder "Draw" sein

    private LocalDateTime timeStamp;

    private LocalDateTime startOfGame;

   // private LocalDateTime startTimer;

    @ElementCollection
    private List<String> moveList;

    private boolean streaming;


}

File: Frontend/SchachEP/src/app/Model/ChessBot.ts
import {User} from './User'

export class ChessBot{
  chessBotId:number|undefined;

  player:User|undefined;

  chessGameName:string|undefined;

  timerUser:number|undefined;

  started:boolean|undefined;

  finished:boolean|undefined;

  playerInGame:boolean|undefined;

  boardState:string|undefined;

  level:number|undefined; // 1, 2 oder 3

  turn:number|undefined;  // 0 -> User, 1 -> Bot

  winner:string|undefined;

  timeStamp:string|undefined;

  startOfGame:Date|undefined;

moveList:string[]|undefined;

constructor() {
  this.chessBotId=-1;
  this.chessGameName='';
  this.started=false;
  this.playerInGame=false;
  this.startOfGame=new Date();
  this.timerUser=-1;
  this.finished=false;
  this.turn=-1;
  this.boardState='';
  this.winner='';
  this.player=new User();
  this.level=-1;
  this.moveList=[];
  this.timeStamp='';
}
}
